Semi-Numerical Exchange (#2567)
psi4/psi4: doc/sphinxman/source/scf.rst

basis set defined for all atoms in the system, or set |scf__df_scf_guess|
to false, which disables this acceleration entirely.

COSX Exchange
~~~~~~~~~~~~~

> The semi-numerical COSX algorithm (described in [Neese:2009:98]_) is so-called
> because the two-electron ERIs are evaluated analytically over one electron 
> coordinate and numerically over the other electron coordinate. The numerical 

Reviewer: ```suggestion
The semi-numerical COSX algorithm (described in [Neese:2009:98]_) evaluates two-electron ERIs
analytically over one electron coordinate and numerically over the other electron coordinate. The numerical
```

Reviewer: One could also mention the connection to the pseudospectral method by Friesner here?

Author: Added a sentence about Friesner

integration is performed on standard DFT quadrature grids, which are described
in :ref:`DFT`. Both the accuracy of the COSX algorithm and also the computational
cost are directly determined by the size of the integration grid, so selection
of the grid is important. This COSX implementation uses two separate grids.
The SCF algorithm is first converged on a smaller grid, followed by a final SCF
iteration on a larger grid. The size of the initial grid is controlled by the
keywords |scf__cosx_radial_points_initial| and |scf__cosx_spherical_points_initial|. The final
grid  is controlled by |scf__cosx_radial_points_final| and
|scf__cosx_spherical_points_final|.

> Defaults have been chosen for both grids that balance cost and accuracy.

Reviewer: ```suggestion
The defaults for both grids aim to balance cost and accuracy.
```

Screening thresholds over integrals, densities, and basis extents are set
with the |scf__cosx_ints_tolerance|, |scf__cosx_density_tolerance|, and

> |scf__cosx_basis_tolerance| keywords, respectively. The |scf__cosx_incfock|

Reviewer: I'm not sure `cosx_basis_tolerance` is a descriptive keyword?

Author: That's fair. I added some more information about the screening keywords

basis set defined for all atoms in the system, or set |scf__df_scf_guess|
to false, which disables this acceleration entirely.

COSX Exchange
~~~~~~~~~~~~~

The semi-numerical COSX algorithm (described in [Neese:2009:98]_) evaluates
two-electron ERIs analytically over one electron coordinate and numerically
over the other electron coordinate. COSX is related to the earlier pseudospectral

> methods of Friesner. In COSX, numerical integration is performed on standard

Reviewer: Why?

> methods of Friesner. In COSX, numerical integration is performed on standard

Reviewer 2: Such historical details are inappropriate for a user guide.

> methods of Friesner. In COSX, numerical integration is performed on standard

Reviewer: I disagree. It's one sentence.

> methods of Friesner. In COSX, numerical integration is performed on standard

Reviewer: Users who read the documentation, meaning people who want to learn about the method. Upon another read, I agree that the organization can be improved; I'll add a suggestion

> methods of Friesner. In COSX, numerical integration is performed on standard

Reviewer: See below

> methods of Friesner. In COSX, numerical integration is performed on standard

Reviewer 3: I generally favor clues that connect code keywords and the literature. Even if it's a distraction at initial get-things-working read, it's potentially reassuring at the what-exactly-have-I-run-and-why stage.

DFT quadrature grids, which are described in :ref:`DFT`.
Both the accuracy of the COSX algorithm and also the computational
cost are directly determined by the size of the integration grid, so selection
of the grid is important. This COSX implementation uses two separate grids.

> The SCF algorithm is first converged on a smaller grid, followed by a final SCF

Author: That is correct. Would it be more clear if I added an additional sentence?:

"This COSX implementation uses two separate grids. The SCF algorithm is first converged on a smaller grid, followed by a final SCF iteration on a larger grid. This results in numerical errors comparable to performing the entire SCF on the expensive larger grid at a computational cost much closer to the smaller grid."

> The SCF algorithm is first converged on a smaller grid, followed by a final SCF

Reviewer 3: I like the additional sentence.

> The SCF algorithm is first converged on a smaller grid, followed by a final SCF

Reviewer 2: Yes - explicitly saying the purpose of the two grid approach is much cleaner.

basis set defined for all atoms in the system, or set |scf__df_scf_guess|
to false, which disables this acceleration entirely.

COSX Exchange
~~~~~~~~~~~~~

> The semi-numerical COSX algorithm (described in [Neese:2009:98]_) evaluates
> two-electron ERIs analytically over one electron coordinate and numerically
> over the other electron coordinate. COSX is related to the earlier pseudospectral
> methods of Friesner. In COSX, numerical integration is performed on standard

Reviewer: ```suggestion
The semi-numerical COSX algorithm described in [Neese:2009:98]_ evaluates
two-electron ERIs analytically over one electron coordinate and numerically
over the other electron coordinate, and belongs to the family of pseudospectral 
methods originally suggested by Friesner. In COSX, numerical integration is performed on standard
```

> methods originally suggested by Friesner. In COSX, numerical integration is performed on standard

Reviewer 2: I still don't agree with mentioning pseudospectral methods and Friesner. Non-experts have no clue who Friesner is and would be better served by reading the Neese paper. Experts already know this. For this reason, in neither case are we helping people who want to learn about the method or people in the "what-exactly-have-I-run-and-why stage."
